build/xtask/src/dist.rs

```diff
@@ -957,17 +957,7 @@ fn build(
         cmd.arg(features.join(","));
     }
 
-    let mut cargo_out = cargo_output_dir(target, path)?;
-
-    // the target dir for each project is set to the same dir, but it ends up
-    // looking something like:
-    //
-    // foo/bar/../target
-    // foo/baz/../target
-    //
-    // these resolve to the same dirs but since their text changes, this causes
-    // rebuilds. by canonicalizing it, you get foo/target for every one.
-    let canonical_cargo_out_dir = fs::canonicalize(&cargo_out)?;
```

Comment:
it might be nice to have a comment here explaining that we are able to do this because we do not set `CARGO_TARGET` and if that changes, this will need to change.